docs/articles/documentation/test-api/selecting-page-elements/selectors.md

```diff
@@ -123,221 +131,306 @@ You can initialize a selector with any of these objects.
         // different selector. The new selector will be initialized with the
         // same function as the old one and with hard-coded parameter values
         // as in the previous example.
-        const submitButton = Selector(elementWithId('submit-button'));
+        const submitButton = Selector(elementWithIdOrClassName('main-element'));
     });
     ```
 
-### Initializers that Return Multiple Nodes
+## Combined Selectors
 
-Functions and CSS selector stings that initialize a selector can return multiple DOM nodes.
-In this instance, you must filter these nodes to select a single node that will eventually be returned by the selector.
-Use the [text](selector-options.md#optionstext) and [index](selector-options.md#optionsindex) options for this.
+The selector API provides methods that can be combined together, thus
+providing you with a flexible functional-style selector mechanism.
 
-The [text](selector-options.md#optionstext) option specifies text content of the node that should be returned
-or a regular expression that matches this content.
+### Filter Multiple DOM Nodes
 
-The [index](selector-options.md#optionsindex) option specifies the index of the node that should be returned.
+If selector returns multiple DOM nodes, you must filter them to select
+a single node that will eventually be returned by the selector.
+The selector provides methods to filter DOM nodes by their index or text.
 
-The following example demonstrates how to use the `text` and `index` options.
+#### nth
 
-```js
-import { expect } from 'chai';
-import { Selector } from 'testcafe';
+Method | Type | Description
+------ | ----- | -----
+`nth(index)` | Selector | Creates a selector that returns an element by its index in the matching set.
 
-fixture `My fixture`
-    .page `http://www.example.com/`;
+#### withText
 
-const removeButton = Selector(cl => document.getElementsByClassName(cl), { text: 'Remove' });
-const secondButton = Selector(cl => document.getElementsByClassName(cl), { index: 1 });
+Method | Type | Description
+------ | ----- | -----
+`withText(text)` | Selector | Creates a selector that filters a matching set by the specified text.
+`withText(re)` | Selector | Creates a selector that filters a matching set using the specified regular expression.
 
-test('A shadowed remove button is focused', async t => {
-    expect(await removeButton('shadowed').focused).to.be.true;
-});
+#### filter
+
+Method | Type | Description
+------ | ----- | -----
+`filter(index)` | Selector | Creates a selector that filters a matching set by `index`.
+`filter(cssSelector)` | Selector | Creates a selector that filters a matching set by `cssSelector`.
+`filter(filterFn)` | Selector | Creates a selector that filters a matching set by `filterFn`; `filterFn` is a client function predicate that receives a node.
 
-test('The second disabled button is visible', async t => {
-    expect(await secondButton('disabled').visible).to.be.true;
+```js
```

Comment:
Should I add a link to this subtitle in the top of the article [here](https://github.com/DevExpress/testcafe/pull/999/files#diff-04c75db5c1c5b1432af03acff5942b96R26) ?